rpc/src/rpc.rs

```diff
@@ -3539,7 +3539,9 @@ pub mod rpc_full {
                     return Err(Error::invalid_params("base58 encoding not supported"));
                 }
 
-                if config_accounts.addresses.len() > post_simulation_accounts.len() {
+                if result.is_ok()
```

Comment:
The `verify_pubkey(&address_str)` call below, which performs a slow base58 decode, is specially what the input account limit is trying to protect from being called thousands of times that could manifest in a malicious request.